modules/tensor_mechanics/src/userobjects/CrystalPlasticityStateVarRateComponentGSS.C

```diff
@@ -59,16 +64,42 @@ CrystalPlasticityStateVarRateComponentGSS::calcStateVariableEvolutionRateCompone
   Real qab;
   Real a = _hprops[3]; // Kalidindi
 
+  // the intial critical resolved shear stress:
+  Real tau_0 = 0.0;
+  // Kalidindi -> tau_0 = 0
+  // classic voce -> tau_0 = tau_y at time 0
+  if (_hprops.size() > 4)
+  {
+    tau_0 = _hprops[4];
+  }
+
+  // compute denomintor of the hardening rule
+  Real delta_tau = tau_sat - tau_0;
+
+  // get the number of coplanar slip systems for BCC or FCC
+  unsigned int nSSperPlane = 0;
+  switch (_crystal_type)
+  {
+    case 0: // FCC12
+      nSSperPlane = 3;
+      break;
+    case 1: // BCC12
+      nSSperPlane = 2;
+      break;
+    default: // error
+      mooseError("Crystal Type Error: Pass valid scalar type ");
+  }
+
   for (unsigned int i = 0; i < _variable_size; ++i)
-    hb(i) = h0 * std::pow(std::abs(1.0 - _mat_prop_state_var[qp][i] / tau_sat), a) *
-            copysign(1.0, 1.0 - _mat_prop_state_var[qp][i] / tau_sat);
+    hb(i) = h0 * std::pow(std::abs(1.0 - (_mat_prop_state_var[qp][i] - tau_0) / delta_tau), a) *
+            copysign(1.0, 1.0 - (_mat_prop_state_var[qp][i] - tau_0) / delta_tau);
```

Comment:
Minor: I know you didn't create this line, but now that I see it in the review, it should be fixed. Would you mind calling the namespaced version of copysign (`std::copysign`). You may need add "#include <cmath>"?